Broaden Swift memory safety to all of WebKit

adetaylor opened the pull request:
#### 7e93e3efdde45416d53639958ae522367bd2f4bd
<pre>
Broaden Swift memory safety to all of WebKit
<a href="https://bugs.webkit.org/show_bug.cgi?id=297589">https://bugs.webkit.org/show_bug.cgi?id=297589</a>
<a href="https://rdar.apple.com/158673557">rdar://158673557</a>

Reviewed by Mike Wyrzykowski and Elliott Williams

This moves the Strict Memory Safety Swift compiler flag from WebKit.xcconfig to
an alternative config file that applies to nearly all WebKit targets.

It also broadens the set of flags we use to match those that WebGPU has
determined to be best.

This is the second attempt to land this change. The first one ran into an error
in older versions of the Swift @Test macro where it emitted unsafe code
(<a href="https://rdar.apple.com/151238560">rdar://151238560</a>). As it happens, WebKit&apos;s continuous integration
infrastructure already has a fix for that in place, but that fix isn&apos;t
being picked up due to some other problem where Swift is sometimes
using an older version of the macro dylib (<a href="https://rdar.apple.com/159858481">rdar://159858481</a>). This needs
to be resolved before we land this commit again. But, those problems
are unrelated to the content of this commit so it&apos;s identical to what
we landed last time.

Canonical link: <a href="https://commits.webkit.org/299897@main">https://commits.webkit.org/299897@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/31e2263574a036f926bf557c2eddc9e2eef00ac7

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/120607 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/40301 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/30953 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/126998 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/72682 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/4bf08014-3081-4778-884b-b58a814728cc) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/122483 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/40998 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/48878 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/91599 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/60860 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/c6bc3f50-7d6a-4992-9fd0-9544dde44cc4) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/123559 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/32740 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/108113 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/72149 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/ce201d45-b81f-4ea8-b66e-6a84db822084) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/31769 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/26218 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/70604 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/102225 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/26400 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/129868 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/47528 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/36080 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/100220 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/47895 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/104294 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/100061 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25402 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/45509 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/23536 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/44187 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/47390 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/53095 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/46858 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/50205 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/48545 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/0baf01db61c708393b0b27e1d22b197f4c43f318. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50359

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/e655dfedc13abee61cb21fd62cb73b665ffecc44. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50359

sheeparegreat reviewed (commented):
LGTM

adetaylor committed:
Broaden Swift memory safety to all of WebKit
https://bugs.webkit.org/show_bug.cgi?id=297589
rdar://158673557

Reviewed by Mike Wyrzykowski and Elliott Williams

This moves the Strict Memory Safety Swift compiler flag from WebKit.xcconfig to
an alternative config file that applies to nearly all WebKit targets.

It also broadens the set of flags we use to match those that WebGPU has
determined to be best.

This is the second attempt to land this change. The first one ran into an error
in older versions of the Swift @Test macro where it emitted unsafe code
(rdar://151238560). As it happens, WebKit's continuous integration
infrastructure already has a fix for that in place, but that fix isn't
being picked up due to some other problem where Swift is sometimes
using an older version of the macro dylib (rdar://159858481). This needs
to be resolved before we land this commit again. But, those problems
are unrelated to the content of this commit so it's identical to what
we landed last time.

Canonical link: https://commits.webkit.org/299897@main
